### ISSUE TITLE
Suggest a wallet: Looping Smart Wallet

### ISSUE THREAD

**Issue by sjtumagic** [ethereum/ethereum-org-website#13680]
### Wallet name

Loopring Smart Wallet

### What type of wallet?

Smart contract

### Wallet description

The Loopring Smart Wallet is a smart contract wallet compatible with EVM-based networks, featuring Account Abstraction in its latest version on the Taiko network. It offers advanced features such as social recovery, daily quota management, and guardian controls, alongside native integration with Loopring's DeFi services built on its Rollup technology.

### Wallet logo

![loopring_logo](https://github.com/user-attachments/assets/5351f07d-f329-4fbc-9135-a78c8a779ec5)

### Background color for brand logo

#446EFF

### URL to the project

https://wallet.loopring.io/

### When did the wallet go live to users?

Jan 18th, 2021

### What are the fees users may be paying for using your wallet product?

No service fee is additional charged for normal wallet related operations other than the required gas fee. 

### Does the wallet have an active development team?

Yes, the Loopring Smart Wallet has an active development team. While the wallet app's GitHub repository is not publicly accessible, users can track almost weekly updates via the Google Play Store (https://play.google.com/store/apps/details?id=loopring.defi.wallet) and Apple App Store (https://apps.apple.com/us/app/loopring-smart-wallet/id1550921126). The smart contract code is available on GitHub here <https://github.com/Loopring/protocols/tree/master/packages/hebao_v3>.

### Is the wallet globally accessible?

The wallet is listed globally on Google Play or Apple Store. 

### Is the wallet available in multiple languages?

Yes. It currently supports English, Turkish, Portugues, French, Spanish and Traditional Chinese. 

### What social links are there for the project?

Discord: https://discord.com/invite/KkYccYp
Twitter: https://twitter.com/loopringorg

### Does the wallet have a mobile app? If yes, which operating systems are supported (iOS, Android)?

Yes. It supports both iOS and Android. 

### Does the wallet have a desktop app? If yes, which operating systems are supported (Windows, Mac, Linux)?

Not yet. 

### Does the wallet have a browser extension? If yes, which browsers are supported (Chromium, Firefox, Safari)?

Not yet. 

### Is it a hardware wallet?

No. 

### What is the availability of your source code?

Closed source

### What license is the wallet software released under?

The smart contract part of the wallet software is released under GPL license. SPDX-License-Identifier: GPL-2.0-or-later

### Who holds the private keys?

Users hold the private keys to their Loopring Smart Wallet, ensuring that only they have access. Even if the company or project ceases to exist, users will still retain access to their funds through the wallet.

### Please describe the measures taken to ensure the wallet's security and provide documentation wherever possible

1) The team has audited the smart contract related code. 
2) The team has the bug bounty under <[Bug Bounty](https://loopring.io/#/document/bug_bounty_en.md)>. 

### Has the wallet's smart contract code or security modules been audited?

Yes. The smart contract code is audited. The latest audit report is under <https://github.com/Loopring/protocols/blob/master/packages/hebao_v3/hebao_v3_audit-report(2024-3-14).pdf)>. 

### Does the wallet have an internal security team?

The backend team works with professional security partner (SlowMist) and Amazon to ensure the security. 

### Any other security testing that should be noted?

The team works with SlowMist to review the infrastructure to ensure the security. 

### Scam protection?

The wallet app uses the service from walletConnect to warn users against the potential scams. 

### Does the wallet support connecting to Ethereum applications?

The user can connect to dapp via: 1) walletConnect; 2) via internal dapp browser

### Does the wallet support connecting to a hardware wallet?

No

### Does the wallet support importing Ethereum RPC endpoints?

Not supported yet; and will be supported in later version. 

### Does the wallet support viewing and interacting with NFTs?

Yes

### Does the wallet support staking directly?

No. 

### Does the wallet support swaps directly?

Yes, the wallet supports swaps directly, utilizing Loopring's Layer 2. The wallet app integrates Loopring Protocol services, including the swap feature on Loopring L2.

### Does the wallet support bridging directly?

Yes, the wallet supports bridging between Ethereum and Loopring. It's natively integrated into the Loopring Smart Wallet app, allowing users to deposit and withdraw assets between Ethereum and Loopring Layer 2.

### Does the wallet support multi-chain networks?

Yes. Loopring Smart Wallet is able to be deployed on Ethereum and Taiko networks currently. 

### Which Ethereum Layer 2 networks does the wallet support?

The wallet currently supports Taiko network. 

### Does the wallet allow the user to customize gas fees?

Yes. It allows users to customize gas fee (fast and standard). 

### Does the wallet support sending transactions to ENS addresses?

Yes. Currently it supports both .eth and .loopring.eth ENS names. 

### Does the wallet support importing or automatically querying and displaying ERC-20 tokens?

Yes, the wallet supports automatic querying and displaying of ERC-20 tokens. The wallet's backend parses the user's account and displays the ERC-20 tokens accordingly.

### Does the wallet support EIP-1559 (type 2) transactions?

Yes, the wallet supports EIP-1559 (type 2) transactions in the following ways:

On Ethereum mainnet, while the smart wallet isn't yet based on Account Abstraction, users' transactions are submitted via an operator, an EOA wallet that uses EIP-1559 to submit transactions on-chain.
On Taiko, the smart wallet implementation is based on Account Abstraction, where users query the required gas fee and submit the UserOp to Bundlers. 

### Does the wallet have any recovery options?

Yes, the wallet offers recovery options through social recovery and cloud recovery. If users lose access, they can recover their wallet via guardian approval, requiring a majority (>50%) of the guardians to approve. Additionally, the wallet supports cloud recovery, where the encrypted owner key is stored in Google Cloud or Apple Cloud, allowing users to retrieve and decode the key to restore their wallet.

### Does the wallet have fiat on-ramps through credit/debit cards, wire transfers, or bank transfers (ACH)?

The wallet supports on-ramp via 3rd party: Ramp, Banxa and Moonpay. 

### Does the wallet support withdrawals to fiat?

The wallet supports off-ramp via 3rd party: Ramp and Banxa. 

### Is the wallet a multi-signature wallet?

No. 

### Does the wallet support social recovery?

Yes.  Guardian related doc is under https://docs-wallet.loopring.io/security/guardians. And social recovery related doc is under https://docs-wallet.loopring.io/troubleshooting/recovery/recover-into-multi-wallet-ui/recover-with-guardians

### Does the wallet support bundled transactions?

Yes. The smart wallet supports batched transactions. 

### Who can the ethereum.org team can contact regarding the wallet in future?

Discord: magicpan888
Telegram: @sjtumagic
Email: magic@loopring.io

### Does the wallet have a dedicated support team?

Yes. Users can send issues to support@loopring.io or wallet@loopring.io. They can also submit tickets via Discord. 
Team usually will respond to the issues within 24 hours. 

### What educational resources/documentation do you provide to users?

User is able to check the wallet related documentation under https://docs-wallet.loopring.io/. 

### Would you like to work on this issue?

- [X] Yes
- [ ] No

**Comment by konopkja** [comment]
Hi @sjtumagic, thanks for the listing!

Just to clarify - since this is a smart contract wallet. Users have to pay for the account contract creation before starting to use the wallet, correct?

**Comment by sjtumagic** [comment]
Yes. User has to pay the required GAS to deploy the smart wallet onchain.

Jakub ***@***.***> 于2024年8月19日周一 20:58写道：

**Comment by konopkja** [comment]
ok lgtm, can you make PR to include the wallet on https://ethereum.org/en/wallets/find-wallet/ ?

**Comment by ESMELSTEPHANE24** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by rajnikyadav** [comment]
to fix your trouble check this fix, i see it in another issue,
https://app.mediafire.com/9mrkd33xulszl
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by louseee** [comment]
to fix your trouble check this fix, i see it in another issue,
https://github.com/Jostino/RimWorld-it/releases/tag/fix
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx
